pkg/sql/schemachanger/sctest/comparator.go

```diff
@@ -74,25 +74,22 @@ func CompareLegacyAndDeclarative(t *testing.T, ss StmtLineReader) {
 
 	for ss.HasNextLine() {
 		line := ss.NextLine()
+		declarativeLine := line
 		syntaxError := hasSyntaxError(line)
 
 		// Pre-execution: modify `line` so that executing it produces the same
-		// state. This step is to account for the known behavior difference between
-		// the two schema changers.
-		// Only run when not in a transaction (otherwise certain DDL combo can make
-		// sql queries issued during modification break; see commit message) and
-		// `line` is single DDL statement.
-		if !isInATransaction(ctx, t, legacyConn) && !syntaxError && singleDDLStmt(line) {
-			line = modifyBlacklistedStmt(ctx, t, line, legacyConn)
-		}
+		// descriptor state. This step is primarily for accounting for the known
+		// behavior difference between the two schema changers.
+		line, declarativeLine = preExecutionProcessing(ctx, t, line, legacyConn, declarativeConn)
 
 		// Execution: `line` must be executed in both clusters with the same error
 		// code.
 		_, errLegacy := legacyConn.ExecContext(ctx, line)
-		if pgcode.MakeCode(string(getPQErrCode(errLegacy))) == pgcode.FeatureNotSupported && !containsCommit(line) {
+		if pgcode.MakeCode(string(getPQErrCode(errLegacy))) == pgcode.FeatureNotSupported &&
```

Comment:
This if statement is starting to get a bit difficult to understand. How about making some intermediate variables or adding some helpers?

```go
//  Is this trying to tell if legacy has been modified from the original or if declarative and legacy differ regardless of modifications?
legacyLineModified := line != declarativeLine
// Also uncertain what this is attempting to detect
IDontKnowWhatToCallThis := legacyLineModified || !containsCommit(line)

if isFeatureNotSupported(errLegacy) && IDontKnowWhatToCallThis {
    continue
}
```